### PR TITLE
Add Uptime KPI + small fixes

### LINKED ISSUE #29: Backend: Implement Uptime KPI
Implement backend logic to create Uptime KPI:

- KPI
- Indicator
- Input
- Logic to generate input (it won't be based on Caddy Logs for this KPI)

| Name | Unique ID | Definition | Value stored in DB for each aggregation | 
|----------|----------|----------|----------|
| Uptime | 2004 | Duration during which the offspot has been turned on | Single value, in minutes |

### PR DESCRIPTION
**Pull request by benoit74**
## Rationale

Fix #29 + small fixes

## Changes

- new KPI `Uptime`
- new Indicator `Uptime`
- new Input `ClockTick`
- new log processing logic to create SharedFilesOperation input sfrom edupi logs in Caddy reverse proxy
- reworked Caddy log lines parsing to use Pydantic (instead of "crappy" home made code)
- small rework of `Period.now()` to return both the `Period` (truncated to the hour) and the full `datetime`

### CODE REVIEW

File: backend/src/offspot_metrics_backend/business/kpis/uptime.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            select(
+                func.sum(IndicatorRecord.value).label("total"),
+            )
+            .join(IndicatorPeriod)
+            .where(IndicatorRecord.indicator_id == UptimeIndicator.unique_id)
+            .where(IndicatorPeriod.timestamp >= start_ts)
+            .where(IndicatorPeriod.timestamp <= stop_ts)
+        ).scalar_one_or_none()
+
+        return UptimeValue(nb_minutes_on=uptime if uptime else 0)
```
Comment (rgaudin) [inline backend/src/offspot_metrics_backend/business/kpis/uptime.py:47]:
```suggestion
        return UptimeValue(nb_minutes_on=uptime or 0)
```

File: backend/src/offspot_metrics_backend/main.py [hunk 2]
```diff
@@ -57,7 +58,7 @@ async def lifespan(self, _: FastAPI):
         if BackendConf.processing_enabled:
             logger.info("Starting processing")
             self.config.parse_configuration()
-            self.processor.startup(current_period=Period.now())
+            self.processor.startup(current_period=Period.now()[0])
```
Comment (rgaudin) [inline backend/src/offspot_metrics_backend/main.py:61]: Could we maybe return a namedtuple from `Period.now()` so it's clearer what we get here?

**Review by rgaudin — CHANGES_REQUESTED** [review]
Good ; I suggested a syntax change somewhere

File: backend/src/offspot_metrics_backend/main.py (hunk 2 above)
Comment (benoit74) [reply backend/src/offspot_metrics_backend/main.py]: Done

**Review by rgaudin — APPROVED** [review]
👍

**Pull request merged**